### PR TITLE
Jenkinsfilecleanups -Lets try deployments

### PR DESCRIPTION
**Pull request by halkeye**

### CODE REVIEW

File: Jenkinsfile_k8s [hunk 1]
```diff
@@ -156,22 +195,22 @@ spec:
...
-          script {
-            def body = """<!-- DEPLOY PREVIEW -->✔️ Deploy Preview for jenkins-plugin-site-pr ready!\n\n😎 Browse the preview: https://deploy-preview-${CHANGE_ID}--jenkins-plugin-site-pr.netlify.app""".stripIndent()
-            def comment = pullRequest.comments.toList().find { it.body.contains("<!-- DEPLOY PREVIEW -->") }
-            if (comment != null) {
-              pullRequest.editComment(comment.id, body)
-            } else {
-              pullRequest.comment(body)
-            }
+
+            sh("""yarn exec netlify deploy --message "Preview deploy for ${CHANGE_ID}" --alias "deploy-preview-${CHANGE_ID}" -d ./plugins/plugin-site/public""")
```
Comment (timja) [inline Jenkinsfile_k8s]: not multiline?

```suggestion
            sh("yarn exec netlify deploy --message "Preview deploy for ${CHANGE_ID}" --alias "deploy-preview-${CHANGE_ID}" -d ./plugins/plugin-site/public")
```

Comment (halkeye) [reply Jenkinsfile_k8s]: You got double quotes inside double quotes. Now thinking about it. I could use single quotes on the outside and bash will handle it.

File: Jenkinsfile_k8s [hunk 2]
```diff
@@ -1,3 +1,39 @@
...
+        "state": status,
+        "environment": environment,
+        "description": description,
+        "log_url": "${env.BUILD_URL}/console",
+        "environment_url": environmentURL
+      ]
+    ),
+    url: "${deployURL}/${id}/statuses"
+  )
+}
```
Comment (dduportal) [inline Jenkinsfile_k8s]: This code is not really following the good practises for pipelines: https://www.jenkins.io/doc/book/pipeline/pipeline-best-practices/#making-sure-to-use-groovy-code-in-pipelines-as-glue .

WDYT avout running this as a groovy script (or whatever other language you would like) on an agent?

**Comment by dduportal** [comment]
For info: added the "deployment" right to the github app https://github.com/jenkins-infra/plugin-site/pull/953/files#diff-a801a318a6ef10e1f344c450dfc3288fe6abfe3092af6ad74093fc6202571a56R2 as requested, hope it solves the last blockers \o/

**Review by timja — APPROVED** [review]

File: Jenkinsfile_k8s [hunk 3]
```diff
@@ -1,3 +1,28 @@
+def recordDeployment(owner, repo, status, environmentURL, environment = "preview", description = "Deploy to preview environment") {
+  withCredentials([usernamePassword(credentialsId: 'github-app-infra', usernameVariable: 'GITHUB_APP', passwordVariable: 'GH_TOKEN')]) {
+    def json = writeJSON(returnText: true, json: [
+        "ref": env.GIT_COMMIT,
+        "environment": environment,
+        "description": description,
+        "required_contexts":[],
+        "auto_merge":false,
+        "transient_environment":false,
```
Comment (timja) [inline Jenkinsfile_k8s]:
```suggestion
        "transient_environment":true,
```

Comment (halkeye) [reply Jenkinsfile_k8s]: see discussion @ https://github.com/jenkins-infra/jenkins.io/pull/4782#discussion_r776079458

**Pull request merged**